shell/components/__tests__/NamespaceFilter.test.ts

```diff
@@ -0,0 +1,213 @@
+import { mount } from '@vue/test-utils';
+import NamespaceFilter from '@shell/components/nav/NamespaceFilter.vue';
+
+describe('component: NamespaceFilter', () => {
+  describe('given namespace select input', () => {
+    it('should be visible', () => {
+      const wrapper = mount(NamespaceFilter, {
+        computed: {
+          filtered: () => [],
+          options:  () => [],
+          value:    () => [],
+        },
+        directives: { shortkey: () => jest.fn() }
+      });
+      const filter = wrapper.find(`[data-testid="namespaces-filter"]`);
+
+      expect(filter).toBeDefined();
+    });
+
+    it('should display no namespace selection', () => {
+      const text = 'none';
+      const wrapper = mount(NamespaceFilter, {
+        computed: {
+          filtered: () => [],
+          options:  () => [],
+          value:    () => [],
+        },
+        mocks:      { $store: { getters: { 'i18n/t': () => text } } },
+        directives: { shortkey: () => jest.fn() }
+      });
+      const element = wrapper.find(`[data-testid="namespaces-values-none"]`).element.textContent;
+
+      expect(element).toContain(text);
+    });
+
+    it('should display the default namespace', () => {
+      const text = 'special namespace';
+      const wrapper = mount(NamespaceFilter, {
+        computed: {
+          filtered: () => [],
+          options:  () => [],
+          value:    () => ([{
+            label: text,
+            kind:  'special',
+          }]),
+        },
+        directives: { shortkey: () => jest.fn() }
+      });
+
+      const element = wrapper.find(`[data-testid="namespaces-values-label"]`).element.textContent;
+
+      expect(element).toContain(text);
+    });
+
+    it('should display the selected namespace', () => {
+      const text = 'current namespace';
+      const wrapper = mount(NamespaceFilter, {
+        computed: {
+          filtered: () => [],
+          options:  () => [],
+          value:    () => [{ label: text }],
+        },
+        directives: { shortkey: () => jest.fn() }
+      });
+
+      const element = wrapper.find(`[data-testid="namespaces-value-0"]`).element.textContent;
+
+      expect(element).toContain(text);
+    });
+
+    it('should display the selected namespace from user preferences if options are available', () => {
+      const text = 'my preference';
+      const key = 'local';
+      const preferences = {
+        [key]: [
+          `ns://${ text }`
+        ]
+      };
+
+      jest.spyOn(NamespaceFilter.computed.value, 'set');
```

Comment:
I get the following warning for `NamespaceFilter.computed.value`

```
Property 'computed' does not exist on type 'VueConstructor<Vue>'.ts(2339)
```

This might be an issue with my environment but warrants calling out.

Reply:
Thanks for verifying. I will chalk it up to my editor.